addons/calendar/models/calendar_alarm.py

```diff
@@ -0,0 +1,304 @@
+                if LastFound:
+                    for alert in LastFound:
+                        all_notif.append(self.do_notif_reminder(alert))
+        return all_notif
+
+    @api.multi
+    def do_mail_reminder(self, alert):
+        res = False
+
+        event = self.env['calendar.event'].browse(alert['event_id'])
+        alarm = self.env['calendar.alarm'].browse(alert['alarm_id'])
+
+        if alarm.alarm_type == 'email':
+            res = self.env['calendar.attendee']._send_mail_to_attendees(
+                [att.id for att in event.attendee_ids],
+                email_from=event.user_id.partner_id.email,
+                template_xmlid='calendar_template_meeting_reminder',
+                force=True
+            )
+
+        return res
+
+    @api.multi
+    def do_notif_reminder(self, alert):
+        alarm = self.env['calendar.alarm'].browse(alert['alarm_id'])
+        event = self.env['calendar.event'].browse(alert['event_id'])
+
+        if alarm.alarm_type == 'notification':
+            message = event.display_time
+
+            delta = alert['notify_at'] - datetime.now()
+            delta = delta.seconds + delta.days * 3600 * 24
+
+            return {
+                'event_id': event.id,
+                'title': event.name,
+                'message': message,
+                'timer': delta,
+                'notify_at': fields.Datetime.to_string(alert['notify_at'])
+            }
+
+
+class CalendarAlarm(models.Model):
+    _name = 'calendar.alarm'
+    _description = 'Event alarm'
+
+    _interval_selection = {'minutes': 'Minute(s)', 'hours': 'Hour(s)', 'days': 'Day(s)'}
+
+    name = fields.Char(required=True)
+    alarm_type = fields.Selection([('notification', 'Notification'), ('email', 'Email')], default='email', oldname='type', required=True)
+    duration = fields.Integer('Amount', default=1, required=True)
+    interval = fields.Selection(list(_interval_selection.iteritems()), string='Unit', default='hours', required=True)
+    duration_minutes = fields.Integer(compute='_get_duration', string='Duration in minutes', store=True, help="Duration in minutes")
+
+    @api.depends('interval', 'duration')
+    def _get_duration(self):
+        for alarm in self:
+            if alarm.interval == "minutes":
+                alarm.duration_minutes = alarm.duration
+            elif alarm.interval == "hours":
+                alarm.duration_minutes = alarm.duration * 60
+            elif alarm.interval == "days":
+                alarm.duration_minutes = alarm.duration * 60 * 24
+            else:
+                alarm.duration_minutes = 0
+
+    @api.onchange('duration', 'interval')
+    def onchange_duration_interval(self):
+        self.display_interval = self._interval_selection.get(self.interval, '')
+        self.name = str(self.duration) + ' ' + self.display_interval
+
+    @api.model
+    def _update_cron(self):
+        try:
+            cron = self.env.ref('calendar.ir_cron_scheduler_alarm')
```

Comment:
set raise_if_not_found to False and remove try..except and at line 280 cron and cron.toggle...